src/streaming/sink.cpp

```diff
@@ -15,4 +25,111 @@ zarr::finalize_sink(std::unique_ptr<zarr::Sink>&& sink)
 
     sink.reset();
     return true;
-}
\ No newline at end of file
+}
+
+std::vector<std::string>
+zarr::construct_data_paths(std::string_view base_path,
```

Comment:
Is this function used anywhere outside the tests? My guess is the actual use comes in a follow-up pr :)

In that case, what's the difference between construct_data_paths and make_dirs? It's hard to tell from the function names. 

I can't tell how construct_data_paths will be used, but is it possible needing to store all the paths in a vector? e.g. Can you pass a path handler as an argument and then this function calls the handler for every constructed path.

Reply:
I see that I missed documenting the return value for `construct_data_paths`. This constructs the paths (which can be used to create directories or S3 object keys), but doesn't actually create the directories.